processor/otel/exceptions_test.go

```diff
@@ -0,0 +1,336 @@
+// limitations under the License.
+
+package otel_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/elastic/apm-server/model"
+)
+
+func TestEncodeSpanEventsNonExceptions(t *testing.T) {
+	nonExceptionEvent := pdata.NewSpanEvent()
+	nonExceptionEvent.SetName("not_exception")
+
+	incompleteExceptionEvent := pdata.NewSpanEvent()
+	incompleteExceptionEvent.SetName("exception")
+	incompleteExceptionEvent.Attributes().InitFromMap(map[string]pdata.AttributeValue{
+		// At least one of exception.message and exception.type is required.
+		conventions.AttributeExceptionStacktrace: pdata.NewAttributeValueString("stacktrace"),
+	})
+
+	_, errors := transformTransactionSpanEvents(t, "java", nonExceptionEvent, incompleteExceptionEvent)
+	require.Empty(t, errors)
+}
+
+func TestEncodeSpanEventsJavaExceptions(t *testing.T) {
+	timestamp := time.Unix(123, 0).UTC()
+
+	exceptionEvent1 := pdata.NewSpanEvent()
+	exceptionEvent1.SetTimestamp(pdata.TimestampUnixNano(timestamp.UnixNano()))
+	exceptionEvent1.SetName("exception")
+	exceptionEvent1.Attributes().InitFromMap(map[string]pdata.AttributeValue{
+		"exception.type":    pdata.NewAttributeValueString("java.net.ConnectException.OSError"),
+		"exception.message": pdata.NewAttributeValueString("Division by zero"),
+		"exception.escaped": pdata.NewAttributeValueBool(true),
+		"exception.stacktrace": pdata.NewAttributeValueString(`
+Exception in thread "main" java.lang.RuntimeException: Test exception
+	at com.example.GenerateTrace.methodB(GenerateTrace.java:13)
+	at com.example.GenerateTrace.methodA(GenerateTrace.java:9)
+	at com.example.GenerateTrace.main(GenerateTrace.java:5)
+	at com.foo.loader/foo@9.0/com.foo.Main.run(Main.java)
+	at com.foo.loader//com.foo.bar.App.run(App.java:12)
+	at java.base/java.lang.Thread.run(Unknown Source)
+`[1:],
+		),
+	})
+	exceptionEvent2 := pdata.NewSpanEvent()
+	exceptionEvent2.SetTimestamp(pdata.TimestampUnixNano(timestamp.UnixNano()))
+	exceptionEvent2.SetName("exception")
+	exceptionEvent2.Attributes().InitFromMap(map[string]pdata.AttributeValue{
+		"exception.type":    pdata.NewAttributeValueString("HighLevelException"),
+		"exception.message": pdata.NewAttributeValueString("MidLevelException: LowLevelException"),
+		"exception.stacktrace": pdata.NewAttributeValueString(`
+HighLevelException: MidLevelException: LowLevelException
+	at Junk.a(Junk.java:13)
+	at Junk.main(Junk.java:4)
+Caused by: MidLevelException: LowLevelException
+	at Junk.c(Junk.java:23)
+	at Junk.b(Junk.java:17)
+	at Junk.a(Junk.java:11)
+	... 1 more
+	Suppressed: java.lang.ArithmeticException: / by zero
+		at Junk.c(Junk.java:25)
+		... 3 more
+Caused by: LowLevelException
+	at Junk.e(Junk.java:37)
+	at Junk.d(Junk.java:34)
+	at Junk.c(Junk.java:21)
+	... 3 more`[1:],
+		),
```

Comment:
Where did you get the format from? Did you instrument some java app with [open-telemetry/opentelemetry-java-instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation)? 

Not a java expert, so maybe the format is a very stable one, but I have following questions:
* assuming you tested with above mentioned library - it is not yet GA (`0.17.0`) - I believe we should add an end-to-end test with an instrumented JAVA app to ensure we realize any breaking changes. 
* is the exception format the same between JAVA versions?
* depending on the answers for above questions - we should ensure to be clear about what is supported in the docs.

Reply:
The format is based on https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html. I'll add a comment to that effect.

> is the exception format the same between JAVA versions?

AFAIK yes, perhaps @elastic/apm-agent-java can comment.

Reply:
FWIW I checked the Java 15 docs and I don't see any significant changes.

Reply:
thanks for checking